language/vm/vm-runtime/src/libra_vm.rs

```diff
@@ -284,13 +284,20 @@ impl LibraVM {
             failed_gas_left = ctx.gas_left();
             let mut gas_free_ctx = SystemExecutionContext::from(ctx);
             self.run_epilogue(&mut gas_free_ctx, txn_data)
-                .and_then(|_| gas_free_ctx.get_transaction_output(txn_data, Ok(())))
+                .and_then(|_| {
+                    gas_free_ctx
+                        .get_transaction_output(txn_data, VMStatus::new(StatusCode::EXECUTED))
+                })
         })
         .unwrap_or_else(|err| {
             let mut gas_free_ctx = SystemExecutionContext::new(remote_cache, failed_gas_left);
-            self.run_epilogue(&mut gas_free_ctx, txn_data)
-                .and_then(|_| gas_free_ctx.get_transaction_output(txn_data, Err(err)))
-                .unwrap_or_else(discard_error_output)
+            match TransactionStatus::from(err) {
```

Comment:
Can you point to the test(s) that fail without the changes to this file?

Reply:
I just updated the PR with those extra tests. I don't like the fact that it's exposing a bunch of internal functions to our e2e test but that seems to be the easiest thing to implement for now.